sextant/src/npk.rs

```diff
@@ -0,0 +1,427 @@
+use anyhow::{anyhow, Context, Result};
+use itertools::Itertools;
+use north::manifest::{Manifest, MountFlag};
+use rand::RngCore;
+use sha2::{Digest, Sha256};
+use sodiumoxide::crypto::{
+    sign,
+    sign::{ed25519::SecretKey, keypair_from_seed, Seed, SEEDBYTES},
+};
+use std::path::Path;
+use std::{
+    fs,
+    fs::{File, OpenOptions},
+    io,
+    io::{BufReader, Read, Seek, SeekFrom::Start, Write},
+    process::Command,
+};
+use tempdir::TempDir;
+use uuid::Uuid;
+
+const DIGEST_SIZE: usize = 32;
+const BLOCK_SIZE: u64 = 4096;
+const UID: u32 = 1000;
+const GID: u32 = 1000;
+
+/// out_dir: where the npk should be packaged to (usually the registry directory)
+pub fn pack(src_dir: &Path, out_dir: &Path, key_file_path: &Path, platform: &str) -> Result<()> {
+    // write platform to manifest
```

Comment:
Add a check that verifies that the squashfs tools are installed, otherwise fail with a descriptive error message